src/api/lib/rabbitmq_bus.rb

```diff
@@ -1,29 +1,42 @@
 class RabbitmqBus
+  cattr_accessor :connection, :exchange
+
   def self.send_to_bus(channel, data)
     publish("#{Configuration.amqp_namespace}.#{channel}", data)
   rescue => e
     Rails.logger.error "Publishing to RabbitMQ failed: #{e.message}"
     Airbrake.notify(e)
+    # reconnect on next message
+    RabbitmqBus.connection.stop
+    RabbitmqBus.exchange = nil
```

Comment:
According to the documentation Bunny does this automatically.

http://rubybunny.info/articles/error_handling.html#initial_rabbitmq_connection_failures

Reply:
the documentation you're pointing to is for initial connections - the publish can happen later on, especially in clockworkd the connection stays open *a long time* and right now you need to restart the clockword to get rabbitmq back on rails (pun intended)

Reply:
It's also done for existing connections:

-> http://rubybunny.info/articles/error_handling.html#network_connection_failures

Sorry for the wrong pointer. I think you're re-implementing bunnies `automatic_recovery`...

Reply:
Yeah, because it doesn't work - I told you, I've seen it broken in action. 

For reference I added an explicit raise on build-test and added debug output to 'connect'

What happens: first time:
```
D, [2018-11-06T07:24:47.677427 #27753] DEBUG -- : [a71974e5-d078-4fc6-bab9-9866cf68a574] [27753:0.14] connect
E, [2018-11-06T07:24:47.712355 #27753] ERROR -- : [a71974e5-d078-4fc6-bab9-9866cf68a574] [27753:0.17] Publishing to RabbitMQ failed: Henne
```

Then on next comments:
```
E, [2018-11-06T07:25:34.269783 #27753] ERROR -- : [ca6cc131-e84f-4ec0-992c-018fab1c012b] [27753:46.73] Publishing to RabbitMQ failed: Henne
```

No reconnect whatsoever.

Reply:
ok, that was broken as I added the exception too high in the stack.

So what I get with more advanced exception throwing:

```
D, [2018-11-06T07:41:23.951362 #31664] DEBUG -- : [31456fd9-16f6-4ef6-bd6d-fecbdfe157d1] [31664:99.50] connect
E, [2018-11-06T07:41:23.970227 #31664] ERROR -- : [31456fd9-16f6-4ef6-bd6d-fecbdfe157d1] [31664:99.52] Got an exception when sending data: Henne (Timeout::Error)

D, [2018-11-06T07:41:33.983635 #31664] DEBUG -- : [31456fd9-16f6-4ef6-bd6d-fecbdfe157d1] [31664:109.54] connect
E, [2018-11-06T07:41:34.008626 #31664] ERROR -- : [31456fd9-16f6-4ef6-bd6d-fecbdfe157d1] [31664:109.56] Got an exception when sending data: Henne (Timeout::Error)
E, [2018-11-06T07:41:34.010000 #31664] ERROR -- : [31456fd9-16f6-4ef6-bd6d-fecbdfe157d1] [31664:109.56] Publishing to RabbitMQ failed: IOError closed stream /usr/lib64/ruby/2.5.0/openssl/buffering.rb:182:in `sysread_nonblock'
```

Reply:
So everything is good, no?